src/lib.rs

```diff
@@ -140,21 +139,94 @@ impl Wiki {
 
         Ok(())
     }
+    /// read all the files from current paths
+    pub fn read_files(&self, file_directory: &str) {
+        //create the default file path
+        if !Path::new(file_directory).exists(){
+            info!("Creating directory for file input: {}.", file_directory);
+            fs::create_dir(file_directory);
+        }
+        for path in self.paths.clone(){
```

Comment:
This clone can be avoided by iterating over the references: `for path in &self.paths`